### ISSUE TITLE
Improve `prefer-string-slice` autofix

### ISSUE THREAD

**Issue by NovemLinguae** [sindresorhus/eslint-plugin-unicorn#2505]
unicorn/prefer-string-slice is good when all the parameters are constants, or when there is only one parameter. But when there are two parameters and they're variables, the autofixes will do things like this:

`editsummary = editsummary.substr(0, editsummary.length - 2); // remove trailing comma`
->
`editsummary = editsummary.slice(0, Math.max(0, editsummary.length - 2)); // remove trailing comma`

I find the Math.max pretty hard to read. I'd prefer that it not autofix that particular sub-pattern. Any interest in adding an option to this rule so that it can be configured to skip that particular sub-pattern when autofixing?

**Comment by edg2s** [comment]
n.b. you should only autofix if the number being substracted is a literal you know to be non-zero.

```
string = string.slice( 0, string.length - y )
```
is usally equal to
```
string = string.slice( 0, -y )
```
but if y is 0 `string.slice( 0, -0 )` gives you the empty string instead of the whole of `string`

**Comment by fregante** [comment]
> Add option for `prefer-string-slice` autofixes to not use `Math.max`

Autofixes should be safe, an option shouldn't change that.

I think this suggestion would be to improve the autofix of this specific scenario from:

```diff
- STR.substr(0, STR.length - NUM)
+ STR.slice(0, Math.max(0, STR.length - NUM))
```

To:

```diff 
- STR.substr(0, STR.length - NUM)
+ STR.slice(0, -NUM)
```

As long as the behavior is always the same.

**Comment by NovemLinguae** [comment]
Agreed that we shouldn't ever do unsafe autofixes. My original idea was to add an option to skip rather than incorrectly fix.

Changing to `STR.slice(0, -NUM)` would be fantastic, assuming it's completely equivalent. Much less verbose.

**Comment by sindresorhus** [comment]
> I think this suggestion would be to improve the autofix of this specific scenario from: [...] As long as the behavior is always the same.

👍
